examples/utils/example-utils/src/modelLoader/instantiateMigratableRuntime.ts

```diff
@@ -0,0 +1,86 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import {
+	IContainer,
+	IContainerContext,
+	IRuntime,
+} from "@fluidframework/container-definitions/internal";
+import {
+	ContainerRuntime,
+	IContainerRuntimeOptions,
+} from "@fluidframework/container-runtime/internal";
+import { IContainerRuntime } from "@fluidframework/container-runtime-definitions/internal";
+import { NamedFluidDataStoreRegistryEntries } from "@fluidframework/runtime-definitions/internal";
+
+import {
+	getDataStoreEntryPoint,
+	type IMigrationTool,
+	MigrationToolFactory,
+} from "../index.js";
+
+/**
+ * @internal
+ */
+export type CreateModelCallback<ModelType> = (
+	runtime: IContainerRuntime,
+	container: IContainer,
+) => Promise<ModelType>;
+
+/**
+ * @internal
+ */
+export interface IMigratableModelContainerRuntimeEntryPoint<T> {
+	getModelAndMigrationTool(
+		container: IContainer,
+	): Promise<{ model: T; migrationTool: IMigrationTool }>;
+}
+
+const migrationToolId = "migration-tool";
+
+const migrationToolRegistryKey = "migration-tool";
+const migrationToolFactory = new MigrationToolFactory();
+
+/**
+ * @internal
+ */
+export const instantiateMigratableRuntime = async <ModelType>(
```

Comment:
This helper is used in place of ModelContainerRuntimeFactory by the customer.

Reply:
Thoughts on creating an interface for the inputs here (i.e. `IMigratableRuntimeProps`)?

Reply:
Could be good longer term, though for initial iteration it's a little easier to just use vanilla parameters.  I'll add a TODO here to consider as we finalize.